src/packages/api/api_manager.py

```diff
@@ -0,0 +1,468 @@
+"""
+api_manager.py - Provide functions for communicating with the Fuel Rats API.
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
+
+import asyncio
+from datetime import datetime
+from html import escape
+import logging
+import json
+import typing
+from uuid import UUID
+
+import aiohttp
+
+from src.config import CONFIG_MARKER
+from src.packages.mark_for_deletion import MarkForDeletion
+from src.packages.rat import Rat
+from src.packages.quotation import Quotation
+from src.packages.rescue import Rescue
+from src.packages.utils import Platforms, Status
+
+LOG = logging.getLogger(f"mecha.{__name__}")
+
+
+class APIManager:
+    """
+    Worker class to interface with the Fuel Rats API.
+    """
+    _config: typing.ClassVar[typing.Dict]
+    _base_url: typing.ClassVar[str]
+    _online_mode: typing.ClassVar[bool]
+    MAX_RETRIES: typing.ClassVar[int] = 3
+    TIMEOUT: typing.ClassVar[aiohttp.ClientTimeout] = aiohttp.ClientTimeout(total=10)
+
+    @classmethod
+    @CONFIG_MARKER
+    def rehash_handler(cls, data: typing.Dict):
+        """
+        Apply new configuration data
+
+        Args:
+            data (Dict): New configuration data
+        """
+        cls._config = data
+        cls._base_url = data['api']['url']
+        cls._online_mode = data['api']['online_mode']
+        if not cls._base_url.endswith('/'):
+            cls._base_url = f"{cls._base_url}/"
+
+    @classmethod
+    @CONFIG_MARKER
+    def validate_config(cls, data: typing.Dict):
+        """
+        Validate the API's configuration values.
+
+        Args:
+            data (Dict): Configuration options
+        """
+        if 'api' not in data.keys():
```

Comment:
`.keys()` is redundant here, `dict.__contains__` checks the keys anyways.
```suggestion
        if 'api' not in data:
```